### PR TITLE
CloseButton

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team. If PR is related to design, please request review from `@beckyrother` in addition._ 

Package: lib-react-components

In anticipation of #1176 and #932.

This makes the close button already in use by the Modal component available as a separate component exported by the shared react components library. Some minor refactoring along the way including making the icon use Grommet icon's `Blank` so that it doesn't have a fixed fill color. 

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-react-components/src/CloseButton/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# CloseButton
+
+A styled Grommet `Button` component that displays a close icon. Has the same API as Grommet's `Button`.
+
+## props
+
+- `closeFn` (function): the function called onClick of the button
+- `className` (string): a string class name
```
Comment (rogerhutchings) [inline packages/lib-react-components/src/CloseButton/README.md]: You shouldn't need `className`, it should be passed down by the base Button component: [link](https://github.com/grommet/grommet/blob/master/src/js/components/Button/Button.js#L121). Have you tried not explicitly passing it in?

File: packages/lib-react-components/src/CloseButton/CloseButton.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  &:hover svg, &:focus svg {
+    opacity: 1;
+  }
+
+  &:disabled {
+    cursor: not-allowed;
+  }
+`
+
+function CloseButton ({ className, closeFn, fill, ...rest }) {
```
Comment (rogerhutchings) [inline packages/lib-react-components/src/CloseButton/CloseButton.js]: Is `fill` used for anything?

File: packages/lib-react-components/src/CloseButton/README.md (hunk 1 above)
Comment (rogerhutchings) [reply packages/lib-react-components/src/CloseButton/README.md]: (it certainly shouldn't be needed by default, but willing to be proved wrong)

File: packages/lib-react-components/src/CloseButton/CloseButton.js (hunk 2 above)
Comment (srallen) [reply packages/lib-react-components/src/CloseButton/CloseButton.js]: Nope. That should be removed.

File: packages/lib-react-components/src/CloseButton/README.md (hunk 1 above)
Comment (srallen) [reply packages/lib-react-components/src/CloseButton/README.md]: I suppose it doesn't have to be explicitly declared.

**Comment by srallen** [comment]
@rogerhutchings this is ready for review again

**Pull request merged**